### PR TITLE
SSCS-9588-ImplementKustomize_SSCS_CoreFrntEnd_Prod

### PR DESCRIPTION
**Pull request by rmurphy-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-9588

### Change description ###
Migration of application SSCS_Core Frontend in SSCS Namespaces to Kustomize.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: k8s/prod/common-overlay/sscs/kustomization.yaml [hunk 1]
```diff
@@ -3,6 +3,8 @@ kind: Kustomization
 namespace: sscs
 bases:
 - ../../../namespaces/sscs
+- ../../../namespaces/sscs/sscs-cor-frontend/sscs-cor-frontend.yaml
```
Comment (dattipoeHMCTS) [inline k8s/prod/common-overlay/sscs/kustomization.yaml:6]: If we're doing the prod deployment should we move this into the main base?

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]
Just double checking the manifests render correctly and are identical

File: k8s/prod/common-overlay/sscs/kustomization.yaml (hunk 1 above)
Comment (rmurphy-hmcts) [reply k8s/prod/common-overlay/sscs/kustomization.yaml]: Files moved into main base.

**Review by dattipoeHMCTS — DISMISSED** [review]

File: k8s/namespaces/sscs/sscs-cor-frontend/demo.yaml [hunk 2]
```diff
@@ -4,26 +4,17 @@ metadata:
...
         EVIDENCE_UPLOAD_QUESTION_PAGE_OVERRIDE_ALLOWED: "true"
-        POST_BULK_SCAN: "true"
-        FT_WELSH: "true"
         HEARING_OUTCOME_TAB_ENABLED: "true"
         MEDIA_FILES_ALLOWED_ENABLED: "true"
         ALLOW_COOKIE_BANNER_ENABLED: "true"
         REQUEST_TAB_ENABLED: "true"
         DUMMY_PROPERTY: true
     global:
       environment: demo
```
Comment (timja) [inline k8s/namespaces/sscs/sscs-cor-frontend/demo.yaml:20]: this should be removed too it's included at the environment level

File: k8s/namespaces/sscs/sscs-cor-frontend/aat.yaml [hunk 3]
```diff
@@ -4,26 +4,17 @@ metadata:
...
         EVIDENCE_UPLOAD_QUESTION_PAGE_OVERRIDE_ALLOWED: "true"
-        POST_BULK_SCAN: "true"
-        FT_WELSH: "true"
         HEARING_OUTCOME_TAB_ENABLED: "true"
         MEDIA_FILES_ALLOWED_ENABLED: "true"
         REQUEST_TAB_ENABLED: "true"
         ALLOW_COOKIE_BANNER_ENABLED: "true"
         DUMMY_PROPERTY: "true"
     global:
       environment: aat
```
Comment (timja) [inline k8s/namespaces/sscs/sscs-cor-frontend/aat.yaml:20]: remove

File: k8s/namespaces/sscs/sscs-cor-frontend/ithc.yaml [hunk 4]
```diff
@@ -4,25 +4,16 @@ metadata:
...
         ADDITIONAL_EVIDENCE_FEATURE_FLAG: "true"
         EVIDENCE_UPLOAD_QUESTION_PAGE_OVERRIDE_ALLOWED: "true"
-        POST_BULK_SCAN: "true"
-        FT_WELSH: "true"
         TEST_PROPERTY: "false"
         HEARING_OUTCOME_TAB_ENABLED: "true"
         REQUEST_TAB_ENABLED: "true"
         MEDIA_FILES_ALLOWED_ENABLED: "true"
     global:
       environment: ithc
```
Comment (timja) [inline k8s/namespaces/sscs/sscs-cor-frontend/ithc.yaml:19]: remove

File: k8s/namespaces/sscs/sscs-cor-frontend/perftest.yaml [hunk 5]
```diff
@@ -4,23 +4,14 @@ metadata:
...
         NODE_ENV: preprod
-        MYA_FEATURE_FLAG: "true"
         ADDITIONAL_EVIDENCE_FEATURE_FLAG: "true"
         EVIDENCE_UPLOAD_QUESTION_PAGE_OVERRIDE_ALLOWED: "true"
-        POST_BULK_SCAN: "true"
-        FT_WELSH: "true"
         HEARING_OUTCOME_TAB_ENABLED: "true"
         MEDIA_FILES_ALLOWED_ENABLED: "true"
     global:
       environment: perftest
```
Comment (timja) [inline k8s/namespaces/sscs/sscs-cor-frontend/perftest.yaml:17]: remove

File: k8s/namespaces/sscs/sscs-cor-frontend/prod.yaml [hunk 6]
```diff
@@ -4,34 +4,18 @@ metadata:
...
         EVIDENCE_UPLOAD_QUESTION_PAGE_OVERRIDE_ALLOWED: "true"
-        POST_BULK_SCAN: "true"
-        FT_WELSH: "true"
         HEARING_OUTCOME_TAB_ENABLED: "true"
         MEDIA_FILES_ALLOWED_ENABLED: "true"
         REQUEST_TAB_ENABLED: "true"
         ALLOW_COOKIE_BANNER_ENABLED: "true"
         DUMMY: true
     global:
       environment: prod
```
Comment (timja) [inline k8s/namespaces/sscs/sscs-cor-frontend/prod.yaml:21]: remove

File: k8s/namespaces/sscs/sscs-cor-frontend/sscs-cor-frontend.yaml [hunk 7]
```diff
@@ -11,4 +11,16 @@ spec:
...
+      replicas: 2
+      useInterpodAntiAffinity: true
+      image: hmctspublic.azurecr.io/sscs-cor/frontend:prod-23f0ef0-20220513104022 #{"$imagepolicy": "flux-system:sscs-cor-frontend"}
+      environment:
+        MYA_FEATURE_FLAG: "true"
+        FT_WELSH: "true"
+        POST_BULK_SCAN: "true"
+    global:
+      tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
+      enableKeyVaults: true
```
Comment (timja) [inline k8s/namespaces/sscs/sscs-cor-frontend/sscs-cor-frontend.yaml:26]: also should be removed, this is set in the env specific helm release

e.g.
https://github.com/hmcts/cnp-flux-config/blob/master/k8s/ithc/ithc-helmrelease.yaml

**Review by timja — COMMENTED** [review]
generally not recommended to do all envs in one PR... but up to you

File: k8s/namespaces/sscs/sscs-cor-frontend/aat.yaml (hunk 3 above)
Comment (rmurphy-hmcts) [reply k8s/namespaces/sscs/sscs-cor-frontend/aat.yaml:20]: Removed

File: k8s/namespaces/sscs/sscs-cor-frontend/ithc.yaml (hunk 4 above)
Comment (rmurphy-hmcts) [reply k8s/namespaces/sscs/sscs-cor-frontend/ithc.yaml:19]: Removed

File: k8s/namespaces/sscs/sscs-cor-frontend/perftest.yaml (hunk 5 above)
Comment (rmurphy-hmcts) [reply k8s/namespaces/sscs/sscs-cor-frontend/perftest.yaml:17]: Removed

File: k8s/namespaces/sscs/sscs-cor-frontend/prod.yaml (hunk 6 above)
Comment (rmurphy-hmcts) [reply k8s/namespaces/sscs/sscs-cor-frontend/prod.yaml:21]: Removed

File: k8s/namespaces/sscs/sscs-cor-frontend/sscs-cor-frontend.yaml (hunk 7 above)
Comment (rmurphy-hmcts) [reply k8s/namespaces/sscs/sscs-cor-frontend/sscs-cor-frontend.yaml:26]: Removed

File: k8s/namespaces/sscs/sscs-cor-frontend/aat.yaml (hunk 3 above)
Comment (timja) [reply k8s/namespaces/sscs/sscs-cor-frontend/aat.yaml:20]: best to not comment till you've pushed unless it's a question :)

**Review by timja — APPROVED** [review]

**Review by dattipoeHMCTS — APPROVED** [review]

**Pull request merged**